tests/test_spectra_matching.py

```diff
@@ -149,4 +147,105 @@ def _send_request(payload: Payload, *args) -> list:
     )
 
     assert len(predictions) == n_spectra
-    assert client._send_request.call_count == n_expected_batches
+    assert client._match_spectra.call_count == n_expected_batches
+
+
+@pytest.fixture
+def mocked_client(mock_send_request):
+    """A usable client with _send_request method mocked."""
+
+    class TestSpectraMatching(SpectraMatching):
+        _algorithm = "test"
+
+    # Token is set so that the class doesn't try to fetch a new token
+    client = TestSpectraMatching(optional_token="token")
+    client._send_request = mock_send_request
+
+    return client
+
+
+def test_spectra_matching_caching(mocked_client, mgf_46_spectra_path):
+    """Sends a request twice. The second one should use cache instead of requesting."""
+
+    input_spectra = list(load_from_mgf(mgf_46_spectra_path))
+
+    spectra = mocked_client.match_spectra(input_spectra, n_best=2)
+
+    assert len(spectra) == 46
+    assert len(mocked_client._cached_results) == len(spectra)
+    assert mocked_client._send_request.call_count == 1
+
+    _ = mocked_client.match_spectra(input_spectra, n_best=2)
+
+    assert mocked_client._send_request.call_count == 1
```

Comment:
I think the assertions do not reflect the explanation in the docstring fully. Maybe we can assert here for `_get_cache_id`called 46 times? so that we make sure all result are got from the cache?

Reply:
ok reading above comment I understand your logic. So  you can ignore this one